### PR TITLE
Purchases: Convert PurchasesList to TS

### PR DESCRIPTION
**Pull request by sirbrillig**
## Proposed Changes

This PR converts the `PurchasesList` component to TypeScript.

## Testing Instructions

This should not make any logic changes so if the tests pass, this should be safe. To be extra sure, visit `/me/purchases` and make sure there are no errors.

### CODE REVIEW

File: client/me/purchases/purchases-list/index.tsx [hunk 1]
```diff
@@ -16,26 +17,51 @@ import NavigationHeader from 'calypso/components/navigation-header';
...
+} from 'calypso/my-sites/checkout/src/hooks/use-stored-payment-methods';
 import { getAllSubscriptions } from 'calypso/state/memberships/subscriptions/selectors';
 import {
 	getUserPurchases,
 	hasLoadedUserPurchasesFromServer,
 	isFetchingUserPurchases,
 } from 'calypso/state/purchases/selectors';
 import getAvailableConciergeSessions from 'calypso/state/selectors/get-available-concierge-sessions';
-import getConciergeNextAppointment from 'calypso/state/selectors/get-concierge-next-appointment';
+import getConciergeNextAppointment, {
```
Comment (JessBoctor) [inline client/me/purchases/purchases-list/index.tsx:35]: Should `getConiergeNextAppointment` be inside the brackets here?

Comment (JessBoctor) [reply client/me/purchases/purchases-list/index.tsx:35]: `TIL` it shouldn't be if it is the default function from the TS file.

**Review by JessBoctor — APPROVED** [review]
Tests have passed and it makes sense. I checked `me/purchases` and didn't receive any errors on load.

File: client/me/purchases/purchases-list/index.tsx (hunk 1 above)
Comment (sirbrillig) [reply client/me/purchases/purchases-list/index.tsx:35]: Yeah. I've come to believe that we should almost always be using named exports only (see [this SO answer](https://stackoverflow.com/questions/46913851/why-and-when-to-use-default-export-over-named-exports-in-es6-modules) for some arguments) but there's a lot of default exports out there and if you need both named and default, this is the syntax we need to use.

**Pull request merged**